exporters/otlp/src/main/java/io/opentelemetry/exporters/otlp/OtlpGrpcMetricExporter.java

```diff
@@ -172,12 +186,67 @@ public Builder setDeadlineMs(long deadlineMs) {
       return this;
     }
 
+    /**
+     * Sets the OTLP endpoint to connect to. Optional.
+     *
+     * @param endpoint endpoint to connect to
+     * @return this builder's instance
+     */
+    public Builder setEndpoint(String endpoint) {
+      this.endpoint = endpoint;
+      return this;
+    }
+
+    /**
+     * Sets use or not TLS, default is false. Optional. Applicable only if {@link Builder#endpoint}
+     * is set to build channel.
+     *
+     * @param useTls use TLS or not
+     * @return this builder's instance
+     */
+    public Builder setUseTls(boolean useTls) {
+      this.useTls = useTls;
+      return this;
+    }
+
+    /**
+     * Add header to request. Optional. Applicable only if {@link Builder#endpoint} is set to build
+     * channel.
+     *
+     * @param key header key
+     * @param value header value
+     * @return this builder's instance
+     */
+    public Builder addHeader(String key, String value) {
+      if (metadata == null) {
+        metadata = new Metadata();
+      }
+      metadata.put(Metadata.Key.of(key, ASCII_STRING_MARSHALLER), value);
+      return this;
+    }
+
     /**
      * Constructs a new instance of the exporter based on the builder's values.
      *
      * @return a new exporter's instance
      */
     public OtlpGrpcMetricExporter build() {
+      if (endpoint != null) {
+        final ManagedChannelBuilder<?> managedChannelBuilder =
+            ManagedChannelBuilder.forTarget(endpoint);
+
+        if (useTls) {
+          managedChannelBuilder.useTransportSecurity();
+        } else {
+          managedChannelBuilder.usePlaintext();
+        }
+
+        if (metadata != null) {
+          managedChannelBuilder.intercept(MetadataUtils.newAttachHeadersInterceptor(metadata));
+        }
+
+        channel = managedChannelBuilder.build();
+      }
```

Comment:
Provide good error message if channel is null here

Reply:
reworked, no needed now due to default endpoint